### PR TITLE
add table stats for table scanner

### PR DESCRIPTION
**Pull request by ming535**

### CODE REVIEW

File: pkg/inputs/mysqlbatch/position_value.go [hunk 1]
```diff
@@ -302,7 +312,42 @@ func PutCurrentPos(cache position_store.PositionCacheInterface, fullTableName st
 		return errors.Trace(err)
 	}
 
+	// update current position and increment scan count
 	batchPositions.Current[fullTableName] = *pos
+	stats := batchPositions.Stats[fullTableName]
+	stats.ScannedCount++
```
Comment (Ryan-Git) [inline pkg/inputs/mysqlbatch/position_value.go]: put this into after commit hook is better I think. `PutCurrentPos` might be called for other purpose.

File: pkg/inputs/mysqlbatch/mysql_table_scanner.go [hunk 2]
```diff
@@ -431,6 +437,9 @@ func (tableScanner *TableScanner) AfterMsgCommit(msg *core.Msg) error {
 		return errors.Trace(err)
 	}
 
+	if err := IncrementScanCount(tableScanner.positionCache, *msg.InputStreamKey); err != nil {
```
Comment (Ryan-Git) [inline pkg/inputs/mysqlbatch/mysql_table_scanner.go]: the scan count should be saved along with current position. why not change it in the table position `p` and save in `PutCurrentPos`?

Comment (ming535) [reply pkg/inputs/mysqlbatch/mysql_table_scanner.go]: @Ryan-Git changed seiralization format for mysqlbatch.

Not intended to put the scan count into `Current` position.

`Current`, `Max`, `Min` use the same struct to define a position.

Comment (Ryan-Git) [reply pkg/inputs/mysqlbatch/mysql_table_scanner.go]: I mean scan count increasing and `current` updating should be atomic. ok for now.

**Review by Ryan-Git — APPROVED** [review]

**Pull request merged**